Robus/src/robus.c

```diff
@@ -162,122 +158,48 @@ error_return_t Robus_SendMsg(ll_container_t *ll_container, msg_t *msg)
     {
         msg->header.source = ctx.node.node_id;
     }
+
+    // Compute the full message size based on the header size info.
+    if (msg->header.size > MAX_DATA_MSG_SIZE)
+    {
+        data_size = MAX_DATA_MSG_SIZE;
+    }
+    else
+    {
+        data_size = msg->header.size;
+    }
+    uint16_t full_size = sizeof(header_t) + data_size;
     // Add the CRC to the total size of the message
     full_size += 2;
```

Comment:
You can create a macro like `CRC_SIZE` and add it in the declaration
 `uint16_t full_size = sizeof(header_t) + data_size + CRC_SIZE;`